File: frontend/src/types/product.ts
export interface ProductTypeItem {
  name: string;
  size: number;
}

export interface Product {
  id: number;
  name: string;
  description?: string;
  price: number;
  image_url?: string;
  created_at: string; 
  stock: number;
  type?: ProductTypeItem[];
  color?: string;
  gender?: string;
  age?: number;
}

File: backend/schemas/product.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime

class ProductTypeItem(BaseModel):
    name: str
    size: int


class ProductOut(BaseModel):
    id: int
    name: str
    description: Optional[str]
    price: int
    image_url: Optional[str]
    created_at: datetime
    stock: Optional[int] = None
    type: Optional[List[ProductTypeItem]] = None
    color: Optional[str]
    gender: Optional[str]
    age: Optional[int]

    model_config = ConfigDict(from_attributes=True)